inline_img_protocol_autodetect.c

```diff
@@ -0,0 +1,127 @@
+/* The only externally visible symbol exported by this module is
```

Comment:
This code should follow the same style as the code in other units. And I think a better place for this is image.c